compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1937,23 +1933,24 @@ private Boolean validateLhsVar(BLangExpression vRef) {
         return true;
     }
 
-    public void visit(BLangCompoundAssignment compoundAssignment) {
+    public void visit(BLangCompoundAssignment compoundAssignment, AnalyzerData data) {
         BType expType;
         BLangValueExpression varRef = compoundAssignment.varRef;
+        SymbolEnv env = data.env;
 
         // Check whether the variable reference is an function invocation or not.
         boolean isValidVarRef = validateLhsVar(varRef);
         if (isValidVarRef) {
             varRef.isCompoundAssignmentLValue = true;
-            this.typeChecker.checkExpr(varRef, env);
+            this.typeChecker.checkExpr(varRef, env, symTable.noType, data.prevEnvs);
             expType = varRef.getBType();
         } else {
             expType = symTable.semanticError;
         }
 
-        this.typeChecker.checkExpr(compoundAssignment.expr, env);
+        this.typeChecker.checkExpr(compoundAssignment.expr, env, symTable.noType, data.prevEnvs);
```

Comment:
In this case (also in previous change) we can write a new `checkExpr` method which has 3 params(expr, env and prevEnv) and call this method within new method.